controller/array_action/array_mediator_ds8k.py

```diff
@@ -483,8 +483,9 @@ def _create_flashcopy(self, source_volume, target_volume):
             else:
                 raise ex
         if not self.validate_flashcopy(api_flashcopy.id):
-            # #TODO Delete flashcopy CSI-1340
+            self._delete_flashcopy(api_flashcopy.id)
             logger.info("Flashcopy is not in a valid state")
+            raise ValueError
```

Comment:
please including actual vs expected state values in the error.
I would change `validate_flashcopy` to something like `ensure_flashcopy_is_valid` to raise a detailed error inside